ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -72,13 +152,42 @@ class SignalProducerSpec: QuickSpec {
 		}
 
 		describe("SignalProducer.try") {
-			pending("should run the operation once per start()") {
+			it("should run the operation once per start()") {
+				var operationRunTimes = 0
+				let operation: () -> Result<String, NSError> = {
+					operationRunTimes++
+
+					return Result<String, NSError>.Success(Box("OperationValue"))
+				}
+
+				SignalProducer.try(operation).start()
+				SignalProducer.try(operation).start()
+
+				expect(operationRunTimes).to(equal(2))
 			}
 
-			pending("should send the value then complete") {
+			it("should send the value then complete") {
+				let operationReturnValue = "OperationValue"
+				let operation: () -> Result<String, NSError> = {
+					return Result<String, NSError>.Success(Box(operationReturnValue))
+				}
+
+				let signalProducer = SignalProducer.try(operation)
+
+				startSignalProducer(signalProducer, expectSentValue: operationReturnValue, sentError: nil, complete: true)
 			}
 
-			pending("should send the error") {
+			it("should send the error") {
+				let operationError = NSError(domain: "com.reactivecocoa.errordomain", code: 4815, userInfo: nil)
+				let operation: () -> Result<String, NSError> = {
+					return Result<String, NSError>.Failure(Box(operationError))
+				}
+
+				let signalProducer = SignalProducer.try(operation)
+
+				// This is not the current behavior, but should it?
```

Comment:
Oh, I just remembered: in RAC2, a signal either errors or completes, but never both. I guess that makes sense, but from an API usage perspective, it can be weird. For example, if you're just interested in the values, and want to know when it's *done sending values*, you would have to handle both the error and the completion. This can be useful (forcing the client to handle errors), but it would be easier to be able to rely on the completion event always being sent.

Considering this behavior has been this way in RAC for a very long time, it's probably for a very good reason :)